### ISSUE TITLE
[Aptos]: TxSign - Getting invalid signature error when sending sign transaction json output to RPC

### ISSUE THREAD

**Issue by punithbm** [trustwallet/wallet-core#2686]
**Describe the bug**
After signing the transaction in the Aptos getting Invalid transaction: Type: Validation Code: INVALID_SIGNATURE

**To Reproduce**
Test Code 👇 
```
const txDataInput = TW.Aptos.Proto.SigningInput.create({
            chainId: 33,
            sender: "0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30",
            transfer: TW.Aptos.Proto.TransferMessage.create({
                amount: new Long(1000),
                to: "0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30",
            }),
            sequenceNumber: new Long(99),
            expirationTimestampSecs: new Long(3664390082),
            gasUnitPrice: new Long(100),
            maxGasAmount: new Long(3296766),
            privateKey: this.PrivateKey.createWithData(
                this.HexCoding.decode(
                    "0x5d996aa76b3212142792d9130796cd2e11e3c445a93118c08414df4f66bc60ec",
                ),
            ).data(),
        });
const input = TW.Aptos.Proto.SigningInput.encode(txDataInput).finish();
const outputData = this.AnySigner.sign(input, this.CoinType.aptos);
const output = TW.Aptos.Proto.SigningOutput.decode(outputData);
const parseJson = JSON.parse(output.json);
console.log("output", parseJson);
```
Output JSON 👇 

```
{
    "expiration_timestamp_secs": "3664390082",
    "gas_unit_price": "100",
    "max_gas_amount": "3296766",
    "payload": {
        "arguments": [
            "0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30",
            "1000"
        ],
        "function": "0x::aptos_account::transfer",
        "type": "entry_function_payload",
        "type_arguments": []
    },
    "sender": "0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30",
    "sequence_number": "99",
    "signature": {
        "public_key": "0xea526ba1710343d953461ff68641f1b7df5f23b9042ffa2d2a798d3adb3f3d6c",
        "signature": "0x5fb98f8be1a29f0e001d4c1b96fca5e3699b75a5219d5ba42e32131119807284031c855418798e90480ebce6e224c68da7c59dbd7d0f5fb1ed6a670f30992101",
        "type": "ed25519_signature"
    }
}
```

**Expected behaviour**
 Generated function value in the JSON is missing 0x1
 Actual - `0x::aptos_account::transfer`, Expected `0x1::aptos_account::transfer`

**API Error Screenshots Reference**
<img width="1300" alt="Screenshot 2022-10-28 at 5 56 35 PM" src="https://user-images.githubusercontent.com/13044958/198587971-8653f0fd-2c5f-432e-9efb-844a219fe655.png">

After correcting the function value in the request
<img width="1291" alt="Screenshot 2022-10-28 at 5 57 20 PM" src="https://user-images.githubusercontent.com/13044958/198588055-81350e00-2f6a-4a11-8eca-6632fbcca0f7.png">

**Additional context**
Expected JSON mentioned in the test case here - https://github.com/trustwallet/wallet-core/blob/master/tests/chains/Aptos/TWAnySignerTests.cpp 
Its also has an invalid signature issue when trying to post transactions. 

Please check and let me know if anything needs to be updated

**Comment by Milerius** [comment]
Hello, the devnet chain ID is 35 now, you can check it here: https://explorer.aptoslabs.com/?network=devnet

**Comment by punithbm** [comment]
@Milerius  its the same issue when trying with chain ID 35 as well 

```
const txDataInput = TW.Aptos.Proto.SigningInput.create({
            chainId: 35,
            sender: "0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30",
            transfer: TW.Aptos.Proto.TransferMessage.create({
                amount: new Long(1000),
                to: "0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30",
            }),
            sequenceNumber: new Long(99),
            expirationTimestampSecs: new Long(3664390082),
            gasUnitPrice: new Long(100),
            maxGasAmount: new Long(3296766),
            privateKey: this.PrivateKey.createWithData(
                this.HexCoding.decode(
                    "0x5d996aa76b3212142792d9130796cd2e11e3c445a93118c08414df4f66bc60ec",
                ),
            ).data(),
        });
const input = TW.Aptos.Proto.SigningInput.encode(txDataInput).finish();
const outputData = this.AnySigner.sign(input, this.CoinType.aptos);
const output = TW.Aptos.Proto.SigningOutput.decode(outputData);
const parseJson = JSON.parse(output.json);
console.log("output", parseJson);
```

<img width="1301" alt="Screenshot 2022-10-28 at 8 51 07 PM" src="https://user-images.githubusercontent.com/13044958/198674214-5741902f-def2-4b14-910f-352e6066f7ef.png">

Can you please check and confirm?

**Comment by punithbm** [comment]
Faced the same issue while trying with the chain ID 1 in the mainnet as well.  Let me know if you need any more information.

**Comment by Milerius** [comment]
> Faced the same issue while trying with the chain ID 1 in the mainnet as well. Let me know if you need any more information.

Can you please let me know which version of wallet-core are u using and which programming langage is your code snippet?

**Comment by punithbm** [comment]
I am using version 3.0.8 trustwallet npm package. Code snippet shared is from javascript

**Comment by Milerius** [comment]
Indeed @punithbm there was a bug with global / constant variable initialization in wasm - should be fixed by #2689

**Comment by punithbm** [comment]
@Milerius, Thanks for checking this on priority. Noted.
